device_independent_test/handshake.py

```diff
@@ -1,40 +1,53 @@
+from device_independent_test import dimension
 from device_independent_test import entanglement
+from device_independent_test import incompatible_measurement
+from device_independent_test import quantum_communicator
 
 class HandShake():
     # Object interface between the user and the test modules
     # Stores a dispatcher to send to modules
     # Runs both individual and sets of tests
 
-    def __init__(self,communicator: quantum_communicator.QuantumDispatcher):
+    def __init__(self, communicator: quantum_communicator.QuantumDispatcher):
         self.dispatcher = communicator
 
     # Run dimenionality test
-    def dimensionality(self,tolerance,shots):
-        pass
-
+    def dimensionality(self, tolerance, shots):
+        (passed, value) = dimension.run_test(self.dispatcher, tolerance, shots)
+        if passed:
+            print("Passed Dimensionality with value: ", value)
+        else:
+            print("Failed Dimensionality with value: ", value)
+        return (passed, value)
+        
     # Run measurement incompatibility test
-    def measurement_incompatibility(self,tolerance,shots):
-        result = incompatible_measurement.run_test(self.dispatcher,tolerance,shots)
-        if result[0]:
-            print("Passed Measurment Incompatibility with value: ", result[1])
+    def measurement_incompatibility(self, tolerance, shots):
+        (passed, value) = incompatible_measurement.run_test(self.dispatcher, tolerance, shots)
+        if passed:
+            print("Passed Measurment Incompatibility with value: ", value)
         else:
-            print("Failed Measurment Incompatibility with value: ", result[1])
-        return result
+            print("Failed Measurment Incompatibility with value: ", value)
+        return (passed, value)
 
     # Run entanglement test
-    def entanglement(self,tolerance,shots):
-        result = entanglement.run_test(self.dispatcher,tolerance,shots)
-        if result[0]:
-            print("Passed Entanglement with value: ", result[1])
+    def entanglement(self, tolerance, shots):
+        (passed, value) = entanglement.run_test(self.dispatcher, tolerance, shots)
+        if passed:
+            print("Passed Entanglement with value: ", value)
         else:
-            print("Failed Entanglement with value: ", result[1])
-        return result
+            print("Failed Entanglement with value: ", value)
+        return (passed, value)
 
     # Run all tests to verify functioning computer/connection
-    def test_all(self,tolerance,shots):
-        pass
+    def test_all(self, tolerance, shots):
+        dimensionality = self.dimensionality(tolerance, shots)
+        measurement_incompatiblility = self.measurement_incompatibility(tolerance, shots)
+        entanglement = self.entanglement(tolerance, shots)
+
+        if dimensionality[0] & measurement_incompatiblility[0] & entanglement[0]:
+            return True
+        else:
+            return False
```

Comment:
we should return the result of each individual test. Maybe use a dictionary


`{'dimension': (pass, score), measurement_incompatibility: (pass, score), entanglement: (pass, score)}`

Reply:
^ good idea. We may also want to store the results of the full run in the handshake object so that it can accumulate tests over time. Not necessary tho